qiskit/algorithms/amplitude_amplifiers/grover.py

```diff
@@ -228,34 +292,50 @@ def amplify(self, amplification_problem: AmplificationProblem) -> "GroverResult"
             if self._sample_from_iterations:
                 power = np.random.randint(power)
             # Run a grover experiment for a given power of the Grover operator.
-            if self._quantum_instance.is_statevector:
-                qc = self.construct_circuit(amplification_problem, power, measurement=False)
-                circuit_results = self._quantum_instance.execute(qc).get_statevector()
+            if self._sampler is not None:
+                qc = self.construct_circuit(amplification_problem, power, measurement=True)
+                job = self._sampler.run([qc])
+
+                try:
+                    results = job.result()
+                except Exception as exc:
+                    raise AlgorithmError("Sampler job failed.") from exc
```

Comment:
Are we sure that `run()` cannot fail?

Reply:
If `run` failed, It will catch an error in `try...except` like this https://github.com/Qiskit/qiskit-terra/blob/cdfd66ef2ea575d8c88606eaf4580714fba84d18/qiskit/algorithms/state_fidelities/compute_uncompute.py#L132